src/main/java/service/FruitService.java

```diff
@@ -0,0 +1,8 @@
+package service;
+
+import java.util.List;
+import model.FruitTransaction;
+
+public interface FruitService {
+    String getReport(List<FruitTransaction> transactions);
```

Comment:
```suggestion
public interface ReportService {
    String getReport();
```
Getting report and executing transactions is **completely different operations**. You should have another class (TransactionExecutor for example) for executing transactions. Here just create report based on current Storage balance (use dao). You do not need transactions here.